src/components/tl/sharp/tl_sharp_lib.c

```diff
@@ -29,9 +33,15 @@ UCC_CLASS_DEFINE(ucc_tl_sharp_lib_t, ucc_tl_lib_t);
 ucc_status_t ucc_tl_sharp_get_lib_attr(const ucc_base_lib_t *lib, /* NOLINT */
                                        ucc_base_lib_attr_t *base_attr)
 {
-    ucc_tl_lib_attr_t *attr      = ucc_derived_of(base_attr, ucc_tl_lib_attr_t);
+    ucc_tl_sharp_lib_t *sharp_lib = ucc_derived_of(lib, ucc_tl_sharp_lib_t);
+    ucc_tl_lib_attr_t  *attr      = ucc_derived_of(base_attr, ucc_tl_lib_attr_t);
 
-    attr->super.flags            = 0;
+    attr->super.flags = 0;
+    if (lib != NULL) {
```

Comment:
can "lib == NULL " here ?

Reply:
yes, it can according to TL iface. CLs may query TL for some static parameters (e.g. thread mode support) w/o lib object, like e.g. here https://github.com/openucx/ucc/blob/5ba01d7ac23b9e4d5444c7e2a7c2b445714fe506/src/components/cl/basic/cl_basic_lib.c#L39